### PR TITLE
NonForcingConstants.non_forcing_inherits_from? 

### PR DESCRIPTION
**Pull request by ngroman**
### Motivation
See many areas in the codebase where checking `if klass < SomethingExpensive` pulls in unneeded dependencies.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by ngroman** [comment]
WIP: runtime with tests written. Would still need to make the changes to resolver.cc

File: gems/sorbet-runtime/lib/types/non_forcing_constants.rb [hunk 1]
```diff
@@ -6,14 +6,30 @@ module T::NonForcingConstants
...
   T::Sig::WithoutRuntime.sig {params(val: BasicObject, klass: String, package: T.nilable(String)).returns(T::Boolean)}
   def self.non_forcing_is_a?(val, klass, package: nil)
+    method_name = "T::NonForcingConstants.non_forcing_is_a?"
+    found_klass = non_forcing_lookup(klass, method_name)
+    !!(found_klass && found_klass.===(val))
+  end
+
+  T::Sig::WithoutRuntime.sig {params(val: Module, klass: String, package: T.nilable(String)).returns(T::Boolean)}
+  def self.non_forcing_inherits_from?(val, klass, package: nil)
+    method_name = "T::NonForcingConstants.non_forcing_inherits_from?"
```
Comment (dmitry-stripe) [inline gems/sorbet-runtime/lib/types/non_forcing_constants.rb:16]: can we move this allocation outside of method invocation? (make it a constant)

File: gems/sorbet-runtime/lib/types/non_forcing_constants.rb [hunk 2]
```diff
@@ -6,14 +6,30 @@ module T::NonForcingConstants
   # shows up in the hover/completion documentation via LSP.
   T::Sig::WithoutRuntime.sig {params(val: BasicObject, klass: String, package: T.nilable(String)).returns(T::Boolean)}
   def self.non_forcing_is_a?(val, klass, package: nil)
+    method_name = "T::NonForcingConstants.non_forcing_is_a?"
```
Comment (dmitry-stripe) [inline gems/sorbet-runtime/lib/types/non_forcing_constants.rb:9]: ditto

**Comment by aprocter-stripe** [comment]
We have a policy of closing pull requests that have not had any recent activity. If you would still like to see this pull request merged, please feel free to reopen it.

As always, please don't hesitate to reach out here on GitHub or in #internals on [sorbet-slack](https://sorbet.org/slack) if you have any questions.

**Pull request closed without merging**